### PR TITLE
New admin actions for cluster replication feature

### PR DESCRIPTION
**Pull request by donatello**

### CODE REVIEW

**Review by vadmeste — APPROVED** [review]
LGTM

File: iam/policy/admin-action.go [hunk 1]
```diff
@@ -85,6 +85,15 @@ const (
 	// GetUserAdminAction - allows GET permission on user info
 	GetUserAdminAction = "admin:GetUser"
 
+	// Cluster Replicate Actions
+
+	// ClusterReplicateAddAction - allow adding replication clusters
+	ClusterReplicateAddAction = "admin:ClusterReplicateAdd"
```
Comment (vadmeste) [inline iam/policy/admin-action.go:91]: It looks like 'Manage' is better than 'Add' here since this action is already used in multiple cases

**Review by vadmeste — CHANGES_REQUESTED** [review]

File: iam/policy/admin-action.go (hunk 1 above)
Comment (harshavardhana) [reply iam/policy/admin-action.go:91]: Yeah makes sense @vadmeste

Comment (donatello) [reply iam/policy/admin-action.go:91]: We use a single set of service credentials to perform all replication management actions on peers such as creating/removing bucket, making service account updates etc. This service account would likely need only one special permission to perform all these actions - I wasn't sure of a name for this, but I think we can use `ClusterReplicateManage`. Will send an update to the PRs.

**Comment by donatello** [comment]
I don't expect we will need further permissions - so we can take this in.

File: iam/policy/admin-action.go (hunk 1 above)
Comment (harshavardhana) [reply iam/policy/admin-action.go:91]: Do we still need `Add` if we have Manage? @donatello ?

Comment (donatello) [reply iam/policy/admin-action.go:91]: Yes, add is to add the cluster repl config, manage is to perform internal replication operations via the svc account created for this purpose. Manage permission just isolates actions by this service account.

**Review by harshavardhana — APPROVED** [review]

File: iam/policy/admin-action.go [hunk 2]
```diff
@@ -85,6 +85,18 @@ const (
...
+
+	// SiteReplicationAddAction - allow adding clusters for site-level replication
+	SiteReplicationAddAction = "admin:SiteReplicationAdd"
+	// SiteReplicationDisableAction - allow disabling a cluster from replication
+	SiteReplicationDisableAction = "admin:SiteReplicationDisable"
+	// SiteReplicationInfoAction - allow getting site replication info
+	SiteReplicationInfoAction = "admin:SiteReplicationInfo"
+	// SiteReplicationOperationAction - allow performing site replication
+	// create/update/delete operations to peers
+	SiteReplicationOperationAction = "admin:SiteReplicationManage"
```
Comment (vadmeste) [inline iam/policy/admin-action.go:98]: s/SiteReplicationManage/SiteReplicationOperation/

**Review by vadmeste — APPROVED** [review]

**Review by vadmeste — CHANGES_REQUESTED** [review]
One comment

File: iam/policy/admin-action.go (hunk 2 above)
Comment (harshavardhana) [inline iam/policy/admin-action.go:98]:
```suggestion
	SiteReplicationOperationAction = "admin:SiteReplicationOperation"
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**